### PR TITLE
ref(discover) Simplify column abstractions in discover

### PR DESCRIPTION
**Pull request by markstory**
Reduce the amount of duplication we maintain for column structures and
use types that make the kind of column being worked on easier to
understand and better leverage typescript features.

The `Field` type now contains the URL name and width. While the `Column`
type contains more metadata about the field. Lastly the `TableColumn`
type contains the width and the `Column` but does not duplicate the
column attributes like it did before.

I've also refactored the resize action handler to use a simpler
eventView function that only handles resizing.

Make inputs the same sizes.

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/queryString.tsx [hunk 1]
```diff
@@ -33,12 +33,12 @@ type QueryValue = string | string[] | undefined | null;
 export function appendTagCondition(
   query: QueryValue,
   key: string,
-  value: string
+  value: null | string
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/queryString.tsx]: The old type was lies. In reality we got `null` here sometimes.

**Review by wmak — APPROVED** [review]
Looks Good To Me

File: src/sentry/static/sentry/app/utils/queryString.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/utils/queryString.tsx]: Nice

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Pull request merged**